src/v/cloud_storage/topic_manifest.cc

```diff
@@ -12,21 +12,193 @@
+            return true;
+        case state::expectKey:
+        case state::expectValue:
+            return false;
+        }
+    }
+
+    bool Key(const char* str, rapidjson::SizeType length, bool /*copy*/) {
+        switch (_state) {
+        case state::expectKey:
+            _key = key_string(str, length);
+            _state = state::expectValue;
+            return true;
+        case state::expectManifestStart:
+        case state::expectValue:
+            return false;
+        }
+    }
+
+    bool String(const char* str, rapidjson::SizeType length, bool /*copy*/) {
+        std::string_view sv(str, length);
+        switch (_state) {
+        case state::expectValue:
+            if ("namespace" == _key) {
+                _namespace = model::ns(ss::sstring(sv));
+            } else if ("topic" == _key) {
+                _topic = model::topic(ss::sstring(sv));
+            } else if ("compression" == _key) {
+                model::compression compression_res;
+                if (boost::conversion::try_lexical_convert(
+                      sv, compression_res)) {
+                    _properties.compression = compression_res;
+                } else {
+                    return false;
+                }
+            } else if ("cleanup_policy_bitflags" == _key) {
+                model::cleanup_policy_bitflags cleanup_policy_bitflags_res;
+                if (boost::conversion::try_lexical_convert(
+                      sv, cleanup_policy_bitflags_res)) {
+                    _properties.cleanup_policy_bitflags
+                      = cleanup_policy_bitflags_res;
+                } else {
+                    return false;
+                }
+            } else if ("compaction_strategy" == _key) {
+                model::compaction_strategy compaction_strategy_res;
+                if (boost::conversion::try_lexical_convert(
+                      sv, compaction_strategy_res)) {
+                    _properties.compaction_strategy = compaction_strategy_res;
+                }
+            } else if ("timestamp_type" == _key) {
+                model::timestamp_type timestamp_type_res;
+                if (boost::conversion::try_lexical_convert(
+                      sv, timestamp_type_res)) {
+                    _properties.timestamp_type = timestamp_type_res;
+                }
+            } else {
+                return false;
+            }
+            _state = state::expectKey;
+            return true;
+        case state::expectManifestStart:
+        case state::expectKey:
+            return false;
+        }
+    }
+
+    bool Uint(unsigned u) { return Uint64(u); }
+
+    bool Uint64(uint64_t u) {
+        switch (_state) {
+        case state::expectValue:
+            if ("version" == _key) {
+                _version = u;
+                if (_version != topic_manifest_version) {
+                    throw std::runtime_error(
+                      "topic manifest version not supported");
```

Comment:
Do we want to throw here? It seems to me that parsing code should stick to parsing, and whatever context is handled the decoded json should check that the version is supported. in either case, it will be useful to include the version in the error message for help in debugging.

Reply:
moved this check to `topic_manifest::update`, after parsing is done